lib/auth/sessions.go

```diff
@@ -102,6 +108,49 @@ func (s *Server) CreateAppSession(ctx context.Context, req types.CreateAppSessio
 	if err = s.Identity.UpsertAppSession(ctx, session); err != nil {
 		return nil, trace.Wrap(err)
 	}
+
+	// User metadata from the audit event
+	userMetadata := identity.GetUserMetadata()
+	userMetadata.User = session.GetUser()
+	userMetadata.AWSRoleARN = req.AWSRoleARN
+
+	// Record peer for the address of the requesting connection.
+	p, ok := peer.FromContext(ctx)
```

Comment:
This is going to be address of proxy making request to this auth server, not the client, right? So it's not really equivalent to what we were logging as the remote addr before.

Reply:
Running this on `branch/v9` (and other branches) without this patch, we've got:

```
  "addr.remote": "192.168.1.232:51369",
```

With this patch, we've got

```
  "addr.remote": "127.0.0.1:51419",
```

So it looks like you're right -- I was taking the explanation from https://github.com/gravitational/teleport/issues/9536#issuecomment-1219430266 at face value here.

So given that this isn't quite right, what's the process here? Should I revert the change in the master branch, or backport all of these and then backport the fix?

Reply:
Let's just submit the fix separately to master (no need to reverting anything) and include it with these backports too.